automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala

```diff
@@ -539,6 +539,40 @@ class SamApiSpec extends FreeSpec with BillingFixtures with Matchers with ScalaF
           .getOrElse(Set.empty) should contain theSameElementsAs Set(allUsersGroupEmail.value),
         5.minutes, 5.seconds)
     }
+
+    "should synchronize the all users group for public policies of constrainable resource type" taggedAs Tags.ExcludeInAlpha in {
+      val resourceId = UUID.randomUUID.toString
+      val user1 = UserPool.chooseAdmin
+      val user1AuthToken = user1.makeAuthToken()
+      val user1Proxy = Sam.user.proxyGroup(user1.email)(user1AuthToken)
+      val allUsersGroupEmail = Sam.user.getGroupEmail("All_Users")(user1AuthToken)
+
+      val resourceTypeName = "workspace"
+      val ownerPolicyName = "owner"
+      val readerPolicyName = "reader"
+
+      Sam.user.createResource(resourceTypeName, CreateResourceRequest(resourceId, Map(ownerPolicyName -> AccessPolicyMembership(Set(user1.email), Set.empty, Set("owner")), readerPolicyName -> AccessPolicyMembership(Set.empty, Set.empty, Set("reader"))), Set.empty))(user1AuthToken)
+      register cleanUp Sam.user.deleteResource(resourceTypeName, resourceId)(user1AuthToken)
+
+      val policies = Sam.user.listResourcePolicies(resourceTypeName, resourceId)(user1AuthToken)
+      val ownerPolicy = policies.filter(_.policyName equals ownerPolicyName).last
+      val readerPolicy = policies.filter(_.policyName equals readerPolicyName).last
+
+      Sam.user.syncResourcePolicy(resourceTypeName, resourceId, ownerPolicyName)(user1AuthToken)
+
+      awaitAssert(
```

Comment:
would be nice to add comment saying this line is to validate actually constrained resource and the other assert is to validate constrainable but not actually constrained resource.

also, shall we add a test for public constrained resource?

Reply:
public constrained resources are not allowed which is tested in unit tests